code/포키/src/main/java/chap03/Appliance.java

```diff
@@ -0,0 +1,30 @@
+package chap03;
+
+import java.util.Objects;
+
+public class Appliance {
+    private int energy;
+    private boolean power;
+    private Grid connectedGrid;
+
+    public Appliance(int useEnergy) {
+        this.energy = useEnergy;
+        this.power = false;
+    }
+
+    public void plugIn(Grid grid) {
+        connectedGrid = grid;
+    }
+
+    public void on() throws Exception {
+        if (connectedGrid.getMaxEnergy() < energy) {
+            throw new Exception("전력 부족");
```

Comment:
좀 더 명확한 예외를 던지면 좋을 것 같네요 😄 

```suggestion
            throw new IllegalArgumentException("전력 부족");
```

Reply:
suggestion 사용해주신거 베리굿굿굿입니닷ㅋㅋㅋ!